modules/broker-core/src/main/java/org/wso2/broker/core/transaction/MessageAcknowledgeAction.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.broker.core.transaction;
+
+import org.wso2.broker.core.Message;
+
+/**
+ * Transaction actions to be perform upon acknowledge
+ */
+public class MessageAcknowledgeAction implements BrokerTransaction.Action {
+
+    private Message message;
+
+    public MessageAcknowledgeAction(Message message) {
+        this.message = message;
+    }
+
+    @Override
+    public void postCommit() {
+        //clear up memory
+        message.release();
```

Comment:
I think we don't need to release messages at this point. This will be release as necessary from the broker-core

Reply:
Please find the changes in d1ebb42